QA_NANO_CONTRACT.md

```diff
@@ -0,0 +1,177 @@
+# QA Nano Contract
+
+- CTA: call-to-action
+
+### Activation
+- [ ] Make sure the wallet-service is disabled for the device
+- [ ] Enable the nano contract feature for the device
+- Close the wallet and reopen
+- [ ] Check a toggle button with two options appear in the dashboard head
+	- Tokens
+	- Nano
+
+### Nano Contract Details
+- [ ] Check the option "Tokens" is selected
+- Select the option "Nano Contracts"
+- [ ] Check the Nano Contract component is in focus with a title "No Nano Contracts"
+- [ ] Check the *Nano Contract Details* component has:
+	- a message, and
+	- a call-to-action (CTA) "Register new"
+
+### Register Nano Contract
+- Register a new Nano Contract by tapping over "Register new" on *Nano Contract Details*
+	- It may focus on a component of loading
+- [ ] Check the Nano Contract Registration screen is in focus
+- [ ] Check the Nano Contract ID is an input
+- [ ] Check the Wallet Address is loaded
+- [ ] Check the "REGISTER NANO CONTRACT" button is disabled
+- Type the following Nano Contract ID `00004f1a3b07f0ffc7ced77a19ae6d4407991a15a281c0ddaf3d6e04a56eba78`
+	- It represents a Nano Contract in the `nano-testnet` network
+- [ ] Check the "REGISTER NANO CONTRACT" button is now actionable
+- Tap on "REGISTER NANO CONTRACT" button
+- [ ] Check for a failure feedback modal to appear
+	- You can't register a Nano Contract from another network
+	- If the message "The informed address does not belong to the wallet" is the failure message, it happens when you are using the wallet service. Make sure to disable it and reload the wallet.
+- Dismiss the modal by tapping outside its area
+- Remove the input
+- [ ] Check a validation message of error appear under the input:
+	- It should informe the field is required
+	- it should be in color red
+- [ ] Check the "REGISTER NANO CONTRACT" button is disabled
+- Type any input
+- [ ] Check the validation error message disappear
+
+### Change network
+- Navigate to the *Settings* screen
+- Go to "Network Settings"
+- Understand disclaimer
+- Customize
+- [ ] Check you not see the fields:
+	- "Wallet Service URL (optional)", and
+	- "Wallet Service WS URL (optional)"
+- If you see the mentioned fields check they are empty or do it by your own before following the next instructions
```

Comment:
This instruction should never happen as the fields should be hidden

Reply:
This can happen if Wallet Service is enabled. But I will add a `[NOTE]` prefix to the instruction.